assets/javascripts/discourse/initializers/enable-user-notes.js

```diff
@@ -84,18 +85,23 @@ export default {
           return dec.attach("user-notes-icon");
         }
       });
-
-      api.decorateWidget("post-admin-menu:after", (dec) => {
-        return dec.h(
-          "ul",
-          dec.attach("post-admin-menu-button", {
-            icon: "pencil-alt",
-            label: "user_notes.attach",
-            action: "showUserNotes",
-            secondaryAction: "closeAdminMenu",
-            className: "add-user-note",
-          })
-        );
+      api.addPostAdminMenuButton((attrs) => {
+        return {
+          icon: "pencil-alt",
+          label: "user_notes.attach",
+          action: () => {
+            showUserNotes(
+              store,
+              attrs.user_id,
+              (count) => {
+                userController.set("userNotesCount", count);
```

Comment:
I pushed a commit which should fix your problem:

https://github.com/discourse/discourse-user-notes/pull/85/commits/5d0ae63a6bea11672c87eca7ef0d3c3c924a589f

There are two parts in the fix:

The complexity of `postAdminMenuButton` is that it has to work with widgets while not being a widget itself, and we also tried to not expose any widget API when working with `postAdminMenuButton` to ensure we can keep moving away from widgets in the future without having to worry of people using widgets related function.

In the old code, `sendWidgetAction` was called, which would ultimately trigger a re-render of widgets. `postAdminMenuButton` is doing it for you automatically, but this case is special, because the actions is in two steps:

- you click the button to show the modal
- then you do the real action which should trigger a render, but `postAdminMenuButton` already did it at previous step

This is the implementation of clicking on a button in the post admin menu:

```
  @action
  async extraAction(button) {
    await this.args.close();
    await button.action(this.args.data.post);
    await this.args.data.scheduleRerender();
  }
```

So to fix this, I created a promise which will block `await button.action(this.args.data.post);` and only be resolved when the count has been set, and then the flow will continue and call `await this.args.data.scheduleRerender();`

---

The other fix is to ensure we use the `post` model instance, and not some POJO representation. This is a default of discourse codebase of mixing these when ideally we should only always use model instance.